packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/LocationDetailView.tsx

```diff
@@ -7,74 +7,69 @@ import { LocationDetailViewControls } from './Controls';
 import { CLASS_BASE } from '../constants';
 import { useControl } from '../../context/controls';
 import { useConfig } from '../../context/config';
-import { useHasValueUpdated } from '@aws-amplify/ui-react-core';
-import { isFolderName } from '../../context/actions/types';
 
 export interface LocationDetailView<
   T extends StorageBrowserElements = StorageBrowserElements,
 > extends ViewComponent<LocationDetailViewControls<T>> {}
 
 export const LocationDetailView: LocationDetailView = () => {
   const { getLocationCredentials, region } = useConfig();
-  const [{ location }, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+  const [{ history, location }, handleUpdateState] = useControl({
+    type: 'NAVIGATE',
+  });
+
+  const { permission, scope } = location ?? {};
+  const bucket = history[0];
+
+  const prefix = history[history.length - 1];
 
   const [locationItemsState, handleListLocationItems] = useAction({
     type: 'LIST_LOCATION_ITEMS',
   });
 
-  const { bucket, permission, scope, prefix } = location ?? {};
   const { data, isLoading } = locationItemsState;
 
-  const hasScopeChanged = useHasValueUpdated(scope);
-
-  const listLocationItems = React.useCallback(
-    ({
-      bucket,
-      // uncomment for testing Amplify buckets
-      prefix = 'public/',
-      // uncomment for testing managed auth buckets
-      // prefix = '',
-      scope,
-    }: {
-      bucket: string;
-      prefix: string | undefined;
-      scope: string;
-    }) =>
-      handleListLocationItems({
-        prefix,
-        config: {
-          bucket,
-          credentialsProvider: async () =>
-            await getLocationCredentials({ permission: permission!, scope }),
-          region,
-        },
-        options: { refresh: true, pageSize: 1000 },
-      }),
-    [getLocationCredentials, permission, handleListLocationItems, region]
-  );
-
   React.useEffect(() => {
-    if (bucket && scope && hasScopeChanged) {
-      // TODO: update to exhaustive call
-      listLocationItems({ bucket, prefix, scope });
+    if (!scope || !permission) {
+      return;
     }
-  }, [bucket, hasScopeChanged, prefix, scope, listLocationItems]);
 
-  const hasItems = !!data.items.length;
+    handleListLocationItems({
+      prefix: prefix === bucket ? '' : prefix,
+      config: {
+        bucket,
+        credentialsProvider: async () =>
+          await getLocationCredentials({ permission, scope }),
+        region,
+      },
+      options: { refresh: true },
```

Comment:
do we still need the pageSize of this call to be 1000?

Reply:
Yes, good catch. Got lost in a rebase